bouncer/tests/broker_level_screening.ts

```diff
@@ -0,0 +1,194 @@
+    pauseBtcBlockProduction(true);
+  }
+  await sleep(waitBeforeReport);
+  // Note: The bitcoin core js lib returns the txId in reverse order.
+  // On chain we expect the txId to be in the correct order (like the Bitcoin internal representation).
+  // Because of this we need to reverse the txId before submitting it as tainted.
+  await submitTxAsTainted(hexStringToBytesArray(txId).reverse());
+  await sleep(stopBlockProductionFor);
+  if (stopBlockProductionFor > 0) {
+    pauseBtcBlockProduction(false);
+  }
+  return Promise.resolve(swapParams.channelId.toString());
+}
+
+// -- Test suite for broker level screening --
+//
+// In this tests we are interested in the following scenarios:
+//
+// 1. No boost and early tx report -> Tainted tx is reported early and the swap is refunded.
+// 2. Boost and early tx report -> Tainted tx is reported early and the swap is refunded.
+// 3. Boost and late tx report -> Tainted tx is reported late and the swap is not refunded.
+async function main() {
+  const MILLI_SECS_PER_BLOCK = 6000;
+  const BLOCKS_TO_WAIT = 2;
+
+  // 1. -- Test no boost and early tx report --
+  testBrokerLevelScreening.log('Testing broker level screening with no boost...');
+  let btcRefundAddress = await newAddress('Btc', randomBytes(32).toString('hex'));
+
+  await brokerLevelScreeningTestScenario(
+    '0.2',
+    false,
+    btcRefundAddress,
+    MILLI_SECS_PER_BLOCK * BLOCKS_TO_WAIT,
+  );
+
+  await observeEvent('bitcoinIngressEgress:TaintedTransactionRejected').event;
+  if (!(await observeBtcAddressBalanceChange(btcRefundAddress))) {
+    throw new Error(`Didn't receive funds refund to address ${btcRefundAddress} within timeout!`);
+  }
+
+  testBrokerLevelScreening.log(`Tainted transaction was rejected and refunded 👍.`);
+
+  // 2. -- Test boost and early tx report --
+  testBrokerLevelScreening.log(
+    'Testing broker level screening with boost and a early tx report...',
+  );
+  btcRefundAddress = await newAddress('Btc', randomBytes(32).toString('hex'));
+
+  await brokerLevelScreeningTestScenario(
+    '0.2',
+    true,
+    btcRefundAddress,
+    MILLI_SECS_PER_BLOCK * BLOCKS_TO_WAIT,
+  );
+  await observeEvent('bitcoinIngressEgress:TaintedTransactionRejected').event;
+
+  if (!(await observeBtcAddressBalanceChange(btcRefundAddress))) {
+    throw new Error(`Didn't receive funds refund to address ${btcRefundAddress} within timeout!`);
+  }
+  testBrokerLevelScreening.log(`Tainted transaction was rejected and refunded 👍.`);
+
+  // 3. -- Test boost and late tx report --
+  // Note: We expect the swap to be executed and not refunded because the tainted tx was to late reported.
+  testBrokerLevelScreening.log('Testing broker level screening with boost and a late tx report...');
+  btcRefundAddress = await newAddress('Btc', randomBytes(32).toString('hex'));
+
+  const channelId = await brokerLevelScreeningTestScenario(
+    '0.2',
+    true,
+    btcRefundAddress,
+    0,
+    MILLI_SECS_PER_BLOCK * 2,
```

Comment:
?
```suggestion
    MILLI_SECS_PER_BLOCK * BLOCKS_TO_WAIT,
```

also these are all the same. if we need to wait the same duration, then would just factor that out

Reply:
For now. But maybe not forever. It's likely to change if we maybe want to extend the test suite, modify it or play with different values to see what the results are. We usually define this kind of constants in test suites. Don't see why it should be bad...

Reply:
just unnecessary complexity atm, we can add the parameterisation when we need it (which might be never) - i won't block the the PR because of it, but yeah, is unnecessary